addons/mail/static/src/models/thread/thread.js

```diff
@@ -168,6 +168,20 @@ function factory(dependencies) {
                     data2.isServerPinned = true;
                 }
             }
+            if ('last_message' in data && data.last_message) {
+                if (!data2.messagesAsServerChannel) {
+                    data2.messagesAsServerChannel = [['unlink-all']];
+                }
+                data2.messagesAsServerChannel.push(['insert', { id: data.last_message.id }]);
```

Comment:
I don't see where messagesAsServerChannel is defined, isn't this pushing into undefined?

Reply:
Just copy/pasted the handler of last_message_id that @alexkuhn wrote as it should work almost the same.
I suppose it's initiliazed by passing in the if

Reply:
The initial one was in livechat and the array was potentially created in `super` (depending on values given). Here we are "in super", the conditions is not necessary.

Should just immediately create the array, once at the top of the method, and then fill it appropriately.

Reply:
```
const data2 = {};
data2.messagesAsServerChannel = [];
```

or

```
const data2 = {
    messagesAsServerChannel: [],
};
```

Reply:
2nd option, not repeating variable name